client/src/App.tsx

```diff
@@ -217,6 +220,13 @@ const App = () => {
     rootsRef.current = roots;
   }, [roots]);
 
+  useEffect(() => {
+    if (mcpClient) {
+      const capabilities = mcpClient.getServerCapabilities();
+      setServerCapabilities(capabilities ?? null);
+    }
+  }, [mcpClient]);
```

Comment:
I'm not sure if this will ever happen, since the Connect button will create a new client every time.